### PR TITLE
Feat: Update reputation chart values

### PR DESCRIPTION
**Pull request by mmioana**
## Description

- We want to update the domain reputation percentage shown in the reputation pie chart, excluding the General team from the total reputation. Based on the normalised total reputation value, we now compute the domain percentages and do not rely on the `reputationPercentage` each domain has.

**How it currently works:**
```
Total reputation: 12,500

General:     2,500 reputation (20%) (earnt specifically in General)

We show this in the chart:
Dev:         2,500 reputation (20%)
Design:      2,500 reputation (20%)
Ops:         2,500 reputation (20%)
All other:   2,500 reputation (20%)
```
**How we propose the calculation to work:**
```
Total reputation: 12,500

General:     2,500 reputation (earnt specifically in General)

Total team reputation: 12,500 - 2,500 = 10,000

We show this in the chart:
Dev:         2,500 reputation (25%)
Design:      2,500 reputation (25%)
Ops:         2,500 reputation (25%)
All other:   2,500 reputation (25%)
```
## Testing

TODO: Please ensure the domain reputation percentages are correctly computed.

* Step 1. Make sure to run the `create-data` script
* Step 2. Go to the teams page and note the reputation percentage values for the domains

Screenshot 2024-09-11 at 09 20 15

* Step 3. Go to the dashboard page and check the values displayed in the `Influence by team` widget - they should be slightly increased given we exclude the General team reputation from the total reputation

Screenshot 2024-09-11 at 09 21 31

* Step 4. Now let's create a new colony by running the `create-colony-url` script
* Step 5. On the dashboard page, the `Influence by team` widget should firstly display no reputation

Screenshot 2024-09-11 at 09 24 25

* Step 6. Make sure the reputation toggle is on - http://0.0.0.0:3001/reputation/monitor/toggle
* Step 7. Award `leela` some reputation 

Screenshot 2024-09-11 at 09 28 14

* Step 8. Wait a bit and then refresh the dashboard page - the `Influence by team` widget should display only `General` with 100%

Screenshot 2024-09-11 at 09 27 01

* Step 9. Create 2 more domains (even more if you feel like it)
* Step 10. Transfer some funds from `planex` to the newly created colony
* Step 11. Accept the incoming funds in the newly created colony http://localhost:9091/{COLONY_NAME}/incoming
* Step 12. Transfer some of the funds from General to each of the newly created domains
* Step 13. Make a simple payment to `leela` from the first domain
* Step 14. After a bit, the chart should update and only show the first domain

Screenshot 2024-09-11 at 09 28 14
* Step 15. Make a simple payment to `leela` from the second domain
* Step 16. After a bit, the chart should update and display both domains

Screenshot 2024-09-11 at 09 45 29

* Step 17. If you visit the teams page, note the values are slightly smaller than the ones displayed in the `Influence by team` widget 

Screenshot 2024-09-11 at 09 47 12

Please feel free testing some more use cases regarding the domain reputation updates and check the values 

## Diffs

**Changes** 🏗

* Update the values retrieved by `getTeamReputationChartData`

Resolves [#2656](https://github.com/JoinColony/colonyCDapp/issues/2656)

### CODE REVIEW

**Review by bassgeta — APPROVED** [review]
Nice and straight forward solution, legit :100: 
My team reputation:
image
The chart:
image
After giving some reputation to leela and forwarding the block time:
image
Making a simple payment from a new domain with 11 CREDS:
image
A simple payment from the second domain with 5 CREDS:
image

Let's go :sunglasses:

**Comment by rumzledz** [comment]
There's probably something going wrong on my local @mmioana but I followed your steps all the way to making a Payment for Leela from a Subdomain -- you can sort of see my actions history in the Latest activity table. But for some reason the Influence by team chart isn't updating for me. Let's jump on a call tomorrow because I need your wisdom 🧘 
image

**Review by rdig — CHANGES_REQUESTED** [review]
Tested this and it works exactly as your testing steps, so from that perspective all good.

The only thing that made me reject this is that _(I think)_ there's a bug in the charting software, which makes the donut chart not fill 100% of it's alloted space _(you'll see what I mean in the screenshots below)_


Screenshot from 2024-09-13 09-52-02
Screenshot from 2024-09-13 09-52-55
Screenshot from 2024-09-13 09-55-20
Screenshot from 2024-09-13 10-00-22
Screenshot from 2024-09-13 10-00-43
Screenshot from 2024-09-13 10-03-40
Screenshot from 2024-09-13 10-04-22
Screenshot from 2024-09-13 10-08-30
Screenshot from 2024-09-13 10-09-46
Screenshot from 2024-09-13 10-17-18
Screenshot from 2024-09-13 10-17-59
Screenshot from 2024-09-13 10-21-29
Screenshot from 2024-09-13 10-22-05
Screenshot from 2024-09-13 10-22-41
Screenshot from 2024-09-13 10-23-48
Screenshot from 2024-09-13 10-24-11
Screenshot from 2024-09-13 10-25-30

So this is the example of the chart not rendering fully in it's allotted space, both with a single domain, and with two domain's worth of reputation

Screenshot from 2024-09-13 10-22-05
Screenshot from 2024-09-13 10-22-41
Screenshot from 2024-09-13 10-23-48
Screenshot from 2024-09-13 10-24-11

The way I made it render fully was to to open the create action sidebar _(don't know why that works)_

Personally, I don't think this is an issue, but I'm sure design or product will pick up on this, so might as well fix it now.

**Comment by mmioana** [comment]
@rdig thank you for testing this out 🙏 

So the issue you're mentioning regarding displaying the 0% for All teams was exactly what I treated here
https://github.com/JoinColony/colonyCDapp/compare/75130a20b3240cdc65bc07eac4f8eb30e6ffb638..41e86689b5dfdee29213ba973db7dee3252f1978

And I didn't manage to reproduce it locally anymore 😢 
I know it's a silly question, but do you have the latest branch version?

**Review by Nortsova — APPROVED** [review]
Hey @mmioana , great work! 🙌

After the first payment, the chart was updated and only showed the first domain:
image
After the second payment, the chart showed both domains:
image
After the third payment, the chart was updated with all 3 teams:
image

Team page:
image

Also, I am not sure that it is related to current PR, I found that the tooltip and legend values are rounded differently (the tooltip should also round to 2 decimal places):
image

Regarding @rdig comment, we can filter chart array and if item 0 just don't push it there ✨🙌

And seems nested domain feature will be really fun 😁

File: src/components/v5/frame/ColonyHome/partials/ReputationChart/utils.ts [hunk 1]
```diff
@@ -40,29 +47,45 @@ export const getTeamReputationChartData = (
...
+   * Exclude the General team reputation from the domain reputation percentage computation
+   */
+  const normalisedTotalReputation = domainsWithoutRoot.reduce(
+    (reputation, team) => reputation + Number(team.reputation),
+    0,
+  );
+
   const topTeams = domainsWithoutRoot
+    // Filter out the domains without reputation in order to not display blank spaces in the chart
+    .filter((domain) => !!domain.reputationPercentage)
```
Comment (mmioana) [inline src/components/v5/frame/ColonyHome/partials/ReputationChart/utils.ts:66]: @Nortsova I am wondering if this filter is not enough to omit teams having 0 reputation

**Review by rdig — APPROVED** [review]
Can't replicate my initial issue with the charts so I consider this now resolved. Everything seems to be rendered as expected no matter how many different reputation states I throw at it.

Nicely done!


Screenshot from 2024-09-18 11-04-32
Screenshot from 2024-09-18 11-04-57
Screenshot from 2024-09-18 11-07-41
Screenshot from 2024-09-18 11-10-59
Screenshot from 2024-09-18 11-12-38
Screenshot from 2024-09-18 11-12-50
Screenshot from 2024-09-18 11-16-12

**Pull request merged**